### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by sinkap**
More instructions for compiling the kernel, pahole and clang.  I do think there is a lot of work that needs to be done in the documentation. Will keep adding more pull requests for this.

### CODE REVIEW

File: .github/README.md [hunk 1]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
...
 ```
 
+If you are compiling the kernel for the first time, install some of the packages required:
+
+On a Debian/Ubuntu distribution:
+
+
+```
+sudo apt install -y \
+  qtdeclarative5-dev \
```
Comment (FlorentRevest) [inline .github/README.md]: the dependency to qtdeclarative5-dev does not seem right to me, it's probably qtbase5-dev

File: .github/README.md [hunk 2]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
...
+  pkg-config \
+  bison \
+  flex \
+  libelf-dev \
+  libssl-dev \
+  libcap-dev \
+  llvm
+```
+
+Enable the folliwing config options to ensure for `hawk` to work:
```
Comment (FlorentRevest) [inline .github/README.md]: following*
"to ensure that hawk works"* or "for hawk to work"

File: .github/README.md [hunk 3]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
...
+
+```
+scripts/config \
+  -e DEBUG_ATOMIC_SLEEP \
+  -e KASAN \
+  -e KMEMLEAK \
+  -e PROVE_LOCKING \
+  -e SECURITYFS \
+  -e CGROUP_BPF \
+  -e IKCONFIG_PROC &&
```
Comment (FlorentRevest) [inline .github/README.md:83]: I don't think that one is necessary

File: .github/README.md [hunk 4]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
...
+If you are doing kernel debugging, you may additionally want to enable:
+
+```
+scripts/config \
+  -e DEBUG_ATOMIC_SLEEP \
+  -e KASAN \
+  -e KMEMLEAK \
+  -e PROVE_LOCKING \
+  -e SECURITYFS \
+  -e CGROUP_BPF \
```
Comment (FlorentRevest) [inline .github/README.md]: This one does not help with debugging, it's only useful for additional BPF programs (ex: cgroup_skb)

File: .github/README.md [hunk 5]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
...
+
+If you are doing kernel debugging, you may additionally want to enable:
+
+```
+scripts/config \
+  -e DEBUG_ATOMIC_SLEEP \
+  -e KASAN \
+  -e KMEMLEAK \
+  -e PROVE_LOCKING \
+  -e SECURITYFS \
```
Comment (FlorentRevest) [inline .github/README.md:82]: This only shows a directory under procfs if the bpf lsm is enabled right? is this relaly important ?

File: .github/README.md [hunk 6]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
...
+
+**clang**
+
+```
+sudo apt install -y cmake && \
+git clone https://github.com/llvm/llvm-project.git  && cd llvm-project && \
+mkdir build && \
+cd build && \
+cmake -DLLVM_ENABLE_PROJECTS=clang -DCMAKE_BUILD_TYPE=Release -G "Unix Makefiles" ../llvm && \
+make -j $(nproc) && \
```
Comment (FlorentRevest) [inline .github/README.md]: apparently `make -j` is what the cool kids do these days (same for pahole)

File: .github/README.md [hunk 7]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
 git clone https://github.com/torvalds/linux.git
 ```
 
+If you are compiling the kernel for the first time, install some of the packages required:
+
+On a Debian/Ubuntu distribution:
+
+
```
Comment (FlorentRevest) [inline .github/README.md]: extra empty line alert

File: .github/README.md [hunk 8]
```diff
@@ -39,6 +40,81 @@ cd  # in your home directory
...
+```
+sudo apt install -y \
+  qtbase5-dev \
+  pkg-config \
+  bison \
+  flex \
+  libelf-dev \
+  libssl-dev \
+  libcap-dev \
+  llvm
```
Comment (FlorentRevest) [inline .github/README.md]: You recommend installing llvm but then later get the user to compile their own llvm. I don't think this one is necessary.

**Review by FlorentRevest — CHANGES_REQUESTED** [review]

File: .github/README.md (hunk 4 above)
Comment (sinkap) [reply .github/README.md]: Removed, thanks

File: .github/README.md (hunk 2 above)
Comment (sinkap) [reply .github/README.md]: Done.

File: .github/README.md (hunk 5 above)
Comment (sinkap) [reply .github/README.md:82]: I think it helps in debugging. Sometimes people don't have `CONFIG_LSM` set correctly or an `lsm=` commandline which overrides it.

File: .github/README.md (hunk 6 above)
Comment (sinkap) [reply .github/README.md]: Thanks! Done.

File: .github/README.md (hunk 8 above)
Comment (sinkap) [reply .github/README.md]: Removed.

File: .github/README.md (hunk 7 above)
Comment (sinkap) [reply .github/README.md]: Removed.

**Pull request merged**